### PR TITLE
python3Packages.omegaconf: 1.4.1 -> 2.1.0

### PR DESCRIPTION
**Pull request by bcdarwin**
python3Packages.hydra: 0.11.3 -> 1.1.1

###### Motivation for this change

Overdue package updates.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [NA] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [NA] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134745` at 3263e4d8 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>python38Packages.hydra</li>
    <li>python38Packages.omegaconf</li>
    <li>python39Packages.hydra</li>
    <li>python39Packages.omegaconf</li>
  </ul>
</details>

File: pkgs/development/python-modules/hydra/default.nix [hunk 1]
```diff
@@ -1,24 +1,26 @@
-{ lib, buildPythonPackage, fetchFromGitHub, isPy27, pytest, omegaconf, pathlib2 }:
+{ lib, buildPythonPackage, fetchFromGitHub, isPy27, pytestCheckHook, pythonOlder
+, importlib-resources, omegaconf, jre_headless, antlr4-python3-runtime }:
 
 buildPythonPackage rec {
   pname = "hydra";
-  version = "0.11.3";
+  version = "1.1.0";
+
+  disabled = isPy27;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hydra/default.nix:8]: Please use pythonOlder.

File: pkgs/development/python-modules/hydra/default.nix [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
-    rev = version;
-    sha256 = "0plbls65qfrvvigza3qvy0pwjzgkz8ylpgb1im14k3b125ny41ad";
+    rev = "v${version}";
+    sha256 = "sha256-fpsgz//wHPeT9SQR59ob0ZOAwGFeEyQIONoXxs8s6cM=";
   };
 
-  checkInputs = [ pytest ];
-  propagatedBuildInputs = [ omegaconf ] ++ lib.optional isPy27 pathlib2;
+  nativeBuildInputs = [ jre_headless ];
+  checkInputs = [ pytestCheckHook omegaconf ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hydra/default.nix:18]:
```suggestion
  checkInputs = [ pytestCheckHook ];
```
duped

File: pkgs/development/python-modules/hydra/default.nix [hunk 3]
```diff
@@ -1,24 +1,26 @@
...
+  propagatedBuildInputs = [ omegaconf antlr4-python3-runtime ] ++ lib.optional (pythonOlder "3.9") [ importlib-resources ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hydra/default.nix:19]:
```suggestion
  propagatedBuildInputs = [ omegaconf antlr4-python3-runtime ]
    ++ lib.optional (pythonOlder "3.9") [ importlib-resources ];
```

File: pkgs/development/python-modules/hydra/default.nix [hunk 4]
```diff
@@ -1,24 +1,26 @@
...
 
-  checkPhase = ''
-    runHook preCheck
-    pytest tests/
-    runHook postCheck
-  '';
+  # test environment setup broken under Nix for a few tests:
+  disabledTests = [ "test_bash_completion_with_dot_in_path" "test_install_uninstall" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hydra/default.nix:22]:
```suggestion
  disabledTests = [
    "test_bash_completion_with_dot_in_path"
    "test_install_uninstall"
  ];
```

File: pkgs/development/python-modules/omegaconf/default.nix [hunk 5]
```diff
@@ -1,20 +1,24 @@
...
-, pytest, pytest-runner, pyyaml, six, pathlib2, isPy27 }:
+{ lib, buildPythonPackage, fetchFromGitHub, pytest-runner, pytest-mock
+, pytestCheckHook, pyyaml, isPy27, jre_minimal, antlr4-python3-runtime }:
 
 buildPythonPackage rec {
   pname = "omegaconf";
-  version = "1.4.1";
+  version = "2.1.0";
+
+  disabled = isPy27;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/omegaconf/default.nix:8]: Please use pythonOlder.

File: pkgs/development/python-modules/omegaconf/default.nix [hunk 6]
```diff
@@ -1,20 +1,24 @@
...
-    sha256 = "1vpcdjlq54pm8xmkv2hqm2n1ysvz2a9iqgf55x0w6slrb4595cwb";
+    rev = "v${version}";
+    sha256 = "sha256-0aDlqPXELxQ/lnw4Hd9es8ldYhUP/TacH9AIyaffwnI=";
   };
 
-  checkInputs = [ pytest ];
-  buildInputs = [ pytest-runner ];
-  propagatedBuildInputs = [ pyyaml six ] ++ lib.optional isPy27 pathlib2;
+  checkInputs = [ pytestCheckHook pytest-mock ];
+  nativeBuildInputs = [ jre_minimal pytest-runner ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/omegaconf/default.nix:18]: Please patch out the requirement for pytest-runner. it is useless in nixpkgs.

File: pkgs/development/python-modules/omegaconf/default.nix [hunk 7]
```diff
@@ -1,20 +1,24 @@
...
+  propagatedBuildInputs = [ antlr4-python3-runtime pyyaml ];
+
+  disabledTestPaths = [ "tests/test_pydev_resolver_plugin.py" ];  # needs pydevd - not in Nixpkgs
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/omegaconf/default.nix:21]: We could package it.

**Comment by piegamesde** [comment]
Works for me, tested on NixOS and Arch Linux.

**Comment by bcdarwin** [comment]
Resolved issues and bumped to latest `facebookresearch/hydra` version.

File: pkgs/development/python-modules/omegaconf/default.nix (hunk 7 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/omegaconf/default.nix:25]: I've started but there are some issues with the test suite -- it takes a long time to run and some concurrency tests fail, so leaving this separate for now.

File: pkgs/development/python-modules/hydra/default.nix [hunk 8]
```diff
@@ -1,24 +1,30 @@
...
+    rev = "v${version}";
+    sha256 = "sha256:1svzysrjg47gb6lxx66fzd8wbhpbbsppprpbqssf5aqvhxgay3qk";
   };
 
-  checkInputs = [ pytest ];
-  propagatedBuildInputs = [ omegaconf ] ++ lib.optional isPy27 pathlib2;
+  nativeBuildInputs = [ jre_headless ];
+  checkInputs = [ pytestCheckHook ];
+  propagatedBuildInputs = [ omegaconf antlr4-python3-runtime ]
+    ++ lib.optional (pythonOlder "3.9") [ importlib-resources ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hydra/default.nix:20]:
```suggestion
    ++ lib.optionals (pythonOlder "3.9") [ importlib-resources ];
```

**Pull request merged**